test/core/OriginCore.js

```diff
@@ -0,0 +1,263 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Test: OriginCore.js
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const eventsDecoder = require('../lib/event_decoder.js');
+
+const OriginCore = artifacts.require('OriginCore');
+const MockToken = artifacts.require('MockToken');
+
+contract('OriginCore', async (accounts) => {
+    describe('reporting a block', async () => {
+        let ost;
+        let originCore;
+
+        beforeEach(async () => {
+            ost = await MockToken.new();
+            originCore = await OriginCore.new(3, ost.address);
+        });
+
+        it('should accept a correct block report', async () => {
+            let expectedBlockHash = '0xb59b762b2a1d476556dd6163bc8ec39967c4debec82ee534c0aed7a143939ed2';
+            let blockHeight = 0;
+
+            await ost.approve(originCore.address, 10 ** 18);
+
+            await originCore.reportBlock(
+                expectedBlockHash,
+                blockHeight,
+                0,
+                '0x82ba3527d2433d8a6c50502f341f165955585b1337512e5cd9550eef623c8fd0',
+                '0x82ba3527d2433d8a6c50502f341f165955585b1337512e5cd9550eef623c8fd0'
+            );
+
+            let reportedHashs = await originCore.getReportedBlockHashes.call(blockHeight);
```

Comment:
we can check the event from `reportBlock`

Reply:
The event won't assert that the report has been stored.